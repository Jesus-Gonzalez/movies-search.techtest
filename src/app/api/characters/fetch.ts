import axios from 'axios';

export const fetchCharacters = () => (
  axios({
    url: 'https://swapi.dev/api/people/',
    responseType: 'json'
  })
  .then(response => response.data)
)
