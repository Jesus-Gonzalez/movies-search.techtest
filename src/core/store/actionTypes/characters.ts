export const Characters = {
  Fetch: 'FETCH_CHARACTERS',
  FetchSuccess: 'FETCH_CHARACTERS_SUCCESS',
  FetchError: 'FETCH_CHARACTERS_ERROR',
};
