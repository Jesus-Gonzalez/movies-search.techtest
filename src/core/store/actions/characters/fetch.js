export const fetchCharacters = () => ({
  type: 'FETCH_CHARACTERS'
});
