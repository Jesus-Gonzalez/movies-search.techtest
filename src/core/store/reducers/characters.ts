export interface IHistoryReducerState {
  error: boolean;
  items: ICharacter[];
  loading: boolean;
}

const initialState: IHistoryReducerState = {
  loading: false,
  items: [],
  error: false
};

export default (state = initialState, action: any) => {
  switch (action.type) {
    case 'FETCH_CHARACTERS':
      return Object.assign(
        {},
        state,
        {
          error: false,
          loading: true,
          items: []
        }
      );

    case 'FETCH_CHARACTERS_SUCCESS':
      return Object.assign(
        {},
        state,
        {
          loading: false,
          items: action.characters.results
        }
      );

    case 'FETCH_CHARACTERS_ERROR':
      return Object.assign(
        {},
        state,
        {
          error: true,
          loading: false,
        }
      );

    default:
      return state;
  }
}
