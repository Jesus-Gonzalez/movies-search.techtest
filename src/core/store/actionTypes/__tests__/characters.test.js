// every action type has the same pattern

import { Characters } from '../characters';

it('creates the action', () => {
  expect(Characters)
    .toMatchSnapshot();
});
