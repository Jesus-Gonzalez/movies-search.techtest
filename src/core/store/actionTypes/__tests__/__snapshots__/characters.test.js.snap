// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates the action 1`] = `
Object {
  "Fetch": "FETCH_CHARACTERS",
  "FetchError": "FETCH_CHARACTERS_ERROR",
  "FetchSuccess": "FETCH_CHARACTERS_SUCCESS",
}
`;
